### PR TITLE
return promise.resolve in initApp commands for adding analytics

### PR DESCRIPTION
**Pull request by gsaray**
Instrument A2HS tracking ootb

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1473
**Related PR**: https://github.com/mobify/progressive-web-sdk/pull/832

## Changes
- return promise.resolve in initApp connector (merlins and SFCC)

## Todos
- [x] (if applicable) analytics events instrumented to measure impact of change
- [x] Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes

## How to test-drive this PR
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)

## To fully test this PR 
- see steps to test in https://github.com/mobify/progressive-web-sdk/pull/832

### CODE REVIEW

File: web/app/connectors/_merlins-connector/app/commands.js [hunk 1]
```diff
@@ -106,5 +106,6 @@ export const initApp = () => (dispatch) => {
     dispatch(setWishlistURL(WISHLIST_URL))
     dispatch(setSignInURL(SIGN_IN_URL))
     dispatch(setAccountURL(MY_ACCOUNT_URL))
-    return dispatch(setCartURL(CART_URL))
+    dispatch(setCartURL(CART_URL))
+    return Promise.resolve()
```
Comment (jnurse) [inline web/app/connectors/_merlins-connector/app/commands.js]: Is this change needed for the stub connector too?

Comment (gsaray) [reply web/app/connectors/_merlins-connector/app/commands.js]: no, it's already there in stub connector https://github.com/mobify/platform-scaffold/blob/e6c1ecd3d3f27d96b05f487a2cc014a0bf29f49f/web/app/connectors/_stub-connector/app/commands.js#L94

**Comment by kschmidtdev** [comment]
What is the reasoning behind returning `Promise.resolve()` instead of returning `dispatch()`? I'm curious since it might have implications for AMP.

**Comment by gsaray** [comment]
@kschmidtdev we're adding standalone analytics here https://github.com/mobify/progressive-web-sdk/pull/832/files#diff-e9562605b78a974925c4444a6967234fR33 and not returning `Promise.resolve()` causes this error 
image and breaks analytics events.

**Comment by jnurse** [comment]
@kschmidtdev all of out commands should be returning a promise of some sort, in some cases that's through a dispatch, but not always. Will this cause problems for AMP?

**Comment by kschmidtdev** [comment]
AMP also requires all of our commands to return a promise (eventually), so just checking! I wasn't sure how `dispatch` behaves when returned.
